File: src/types/services.ts
export type ServiceStartType =
  | "Automatic"
  | "AutoDelayed"
  | "Manual"
  | "Disabled"
  | "Unknown";

export type ServiceState =
  | "Running"
  | "Stopped"
  | "StartPending"
  | "StopPending"
  | "Unknown";

export interface ServiceEntry {
  name: string;
  displayName: string;
  description: string;
  startType: ServiceStartType;
  currentState: ServiceState;
  isProtected: boolean;
}

export interface ServiceChange {
  serviceName: string;
  targetStartType: ServiceStartType;
  rationale: string;
}

export interface ServicePreset {
  id: string;
  name: string;
  description: string;
  changes: ServiceChange[];
}

export interface ServicePresetResult {
  serviceName: string;
  success: boolean;
  message: string;
}

File: src-tauri/src/models/services.rs
// Copyright (c) 2026 Seppe Willemsens (ZIPREX420). MIT License.
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
pub enum ServiceStartType {
    Automatic,
    AutoDelayed,
    Manual,
    Disabled,
    Unknown,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
pub enum ServiceState {
    Running,
    Stopped,
    StartPending,
    StopPending,
    Unknown,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ServiceEntry {
    pub name: String,
    pub display_name: String,
    pub description: String,
    pub start_type: ServiceStartType,
    pub current_state: ServiceState,
    pub is_protected: bool,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ServiceChange {
    pub service_name: String,
    pub target_start_type: ServiceStartType,
    pub rationale: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ServicePreset {
    pub id: String,
    pub name: String,
    pub description: String,
    pub changes: Vec<ServiceChange>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ServicePresetResult {
    pub service_name: String,
    pub success: bool,
    pub message: String,
}
